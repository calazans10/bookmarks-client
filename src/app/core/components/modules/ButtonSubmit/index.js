import React from 'react';
import PropTypes from 'prop-types';
import { Button } from './style';

const ButtonSubmit = ({ children }) => <Button type="submit">{children}</Button>;

ButtonSubmit.propTypes = {
  children: PropTypes.node.isRequired,
};

export default ButtonSubmit;
