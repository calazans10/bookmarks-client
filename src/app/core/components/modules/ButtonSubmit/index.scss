@import 'styles/utilities';

.button-submit {
  width: rem(140);
  padding: rem(10);
  border: none;
  background-color: #27ae60;
  color: #fff;
  font-size: em(13);
  font-weight: 500;
  letter-spacing: 2px;
  text-transform: uppercase;
  cursor: pointer;

  &:disabled {
    opacity: 0.7;
  }
}
