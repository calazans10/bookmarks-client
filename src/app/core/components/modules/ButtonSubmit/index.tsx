import React from 'react';
import { Button } from './style';

type ButtonSubmitProps = {
  children: React.ReactNode;
};

const ButtonSubmit = ({ children }: ButtonSubmitProps) => <Button type="submit">{children}</Button>;

export default ButtonSubmit;
