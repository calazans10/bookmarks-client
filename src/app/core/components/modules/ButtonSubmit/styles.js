import styled from 'styled-components';
import { em, rem } from 'polished';

export const Button = styled.button`
  width: ${rem('140px')};
  padding: ${rem('10px')};
  border: none;
  background-color: #27ae60;
  color: #fff;
  font-size: ${em('13px')};
  font-weight: 500;
  letter-spacing: 2px;
  text-transform: uppercase;
  cursor: pointer;

  &:disabled {
    opacity: 0.7;
  }
`;
