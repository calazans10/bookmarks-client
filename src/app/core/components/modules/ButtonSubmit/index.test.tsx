import React from 'react';
import { render } from '@testing-library/react';
import ButtonSubmit from '.';

describe('ButtonSubmit', () => {
  it('renders without crashing', () => {
    const { container } = render(<ButtonSubmit>Save</ButtonSubmit>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
