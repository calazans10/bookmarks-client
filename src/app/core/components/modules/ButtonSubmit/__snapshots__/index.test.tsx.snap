// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonSubmit renders without crashing 1`] = `
<button
  class="sc-AxjAm hNPono"
  type="submit"
>
  Save
</button>
`;
